### PR TITLE
cleanup cli docs generation: BUILD_ALL_VERSION=1

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### CODE REVIEW

File: site2/website-next/docs/reference-cli-broker-tool.md [hunk 1]
```diff
@@ -1,36 +0,0 @@
...
----
-
-## `broker-tool`
-
-The `broker- tool` is used for operations on a specific broker.
-
-Usage
-
-```bash
-broker-tool command
```
Comment (tisonkun) [inline site2/website-next/docs/reference-cli-broker-tool.md:14]: It seems that we remove this utility for a long time.

cc @merlimat @codelipenghui @nicoloboschi do you have ideas of the specific commit we remove this tool? I'm thinking of which versions of docs we should do the corresponding changes.

File: site2/tools/pulsar-config-doc-gen.sh [hunk 2]
```diff
@@ -52,10 +52,3 @@ $JAVA -cp "$CLIENT_CP:$(cat "$f")" $GEN_DOCS_CLIENT -c $CLIENT_CONF.ClientConfig
 $JAVA -cp "$CLIENT_CP:$(cat "$f")" $GEN_DOCS_CLIENT -c $CLIENT_CONF.ProducerConfigurationData > "$CLIENT_DIR"/client-configuration-producer.md
 $JAVA -cp "$CLIENT_CP:$(cat "$f")" $GEN_DOCS_CLIENT -c $CLIENT_CONF.ConsumerConfigurationData > "$CLIENT_DIR"/client-configuration-consumer.md
 $JAVA -cp "$CLIENT_CP:$(cat "$f")" $GEN_DOCS_CLIENT -c $CLIENT_CONF.ReaderConfigurationData > "$CLIENT_DIR"/client-configuration-reader.md
-
-# copy CLI tools docs
-tools="bookkeeper pulsar-daemon pulsar-shell"
-for tool in $tools
-do
-    cp "$WEBSITE/docs/reference-cli-$tool.md" "$VERSIONED_DIR/$tool/$tool.md"
-done
```
Comment (tisonkun) [inline site2/tools/pulsar-config-doc-gen.sh:61]: This code snippet along with the existence of the `reference-cli-*` files causes the current https://pulsar.apache.org/reference/#/next/bookkeeper/bookkeeper page still have a wrongly parsed yaml header.

We need to remove all this stuff to finish what @Anonymitaet proposed at https://github.com/apache/pulsar/pull/18101

File: site2/website-next/docs/reference-cli-broker-tool.md (hunk 1 above)
Comment (tisonkun) [reply site2/website-next/docs/reference-cli-broker-tool.md:14]: OK I found https://github.com/apache/pulsar/pull/18051#issuecomment-1280297019.

File: site2/tools/pulsar-config-doc-gen.sh (hunk 2 above)
Comment (urfreespace) [reply site2/tools/pulsar-config-doc-gen.sh:61]: @SignorMercurio would you pls help to double-check it? thanks.

Comment (SignorMercurio) [reply site2/tools/pulsar-config-doc-gen.sh:61]: Yes, this code snippet is no longer needed and should be removed. So does `reference-cli-*.md`. Thanks @tisonkun!

Comment (Anonymitaet) [reply site2/tools/pulsar-config-doc-gen.sh:61]: OK, thank you all!

**Comment by tisonkun** [comment]
ping @Anonymitaet @urfreespace as a reminder. I think this patch is ready to merge.

**Pull request merged**